laikamilter.py

```diff
@@ -628,15 +636,15 @@ def _zmqGetFlags(self, numRetries, milterContext):
     def _zmqSendBuffer(self, milterContext,numRetries, REQUEST_TIMEOUT,SERVER_ENDPOINT):
         gotResponseFromScanner=-1
         self.client = Client(SERVER_ENDPOINT)
-        
         log = milterContext.uuid+" Sending "+ str(milterContext.qid)+" to "+ SERVER_ENDPOINT
         self.logger.writeLog(syslog.LOG_DEBUG, "%s"%(str(log)))
         myhostname = socket.gethostname()
         externalObject = ExternalObject(
                                         buffer=milterContext.fileBuffer, 
                                         externalVars=ExternalVars(
                                                                   filename=milterContext.archiveFileName, 
-                                                                  source=milterContext.milterConfig.milterName+"-"+str(myhostname[:myhostname.index(".")]),
+                                                                  source=milterContext.milterConfig.milterName+"-"+ \
+                                                                          (str(myhostname[:myhostname.index(".")]) if "." in myhostname else str(myhostname)),
```

Comment:
Your right, I didn't realize splitting on something that might not exist would just throw it into the first entry of the list. That would be a better fix.